### ISSUE TITLE
Type evaluation of expressions using the Null-Safe Operator is wrong

### ISSUE THREAD

**Issue by tlamy** [kalessil/phpinspectionsea#1889]
```php
interface BankIdInterface extends Stringable
{
    public function getBic(): string;
}

…

$bankId = $targetAccount->getBankId(); // returns BankIdInterface|null
if($bankId?->getBic() !== null) {
    …doSomething();
}
```
![image](https://github.com/kalessil/phpinspectionsea/assets/1841537/7d30b13f-bdb0-4493-acf8-bf6843c18542)

Even though `getBic()` always returns non-null, the null-safe operator adds the null return type to the lhs expression. Thus the inspection result is a false positive.

Php Inspections (EA Ultimate) 2023.1
PhpStorm 2023.1.2 / macOS 13.4 (22F66) / Apple M1

PS: I just renewed my Ultimate license after the upload of 2023.1 - hope you tackle all those >=8.0 issues soon.
